src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -255,8 +255,12 @@ public bool IsEmpty
         /// cref="ConcurrentQueue{T}"/>.</returns>
         public T[] ToArray()
         {
-            return ToList().ToArray();
+            List<T> list = ToList();
```

Comment:
> Does this create a list even if empty array will be returned?

Yes.  The changes necessary to avoid that didn't seem worthwhile.  If we really want to, we can do it separately; it would have an impact on some of the synchronization being done, and it would require more scrutiny.